### ISSUE TITLE
Setting NumSamples::One enables multisampling/antialiasing when it should remain disabled

### ISSUE THREAD

**Issue by drojf** [ggez/ggez#552]
Platform: Windows
GPU: Intel HD4600
ggez version: 0.5.0-rc.0

## Testing on 0.5

I migrated my project to 0.5 recently, and had no problems running the project on my newish laptop.

I then tried to run the project on an older computer (Intel HD4600). The program couldn't open the window - glutin failed opening the window with a "NoAvailablePixelFormat" error. 

To perform a 'minimal test', I compiled and ran the 'basic window' Glutin example project.
This ran with no problems. I then enabled antialiasing, which caused the same "NoAvailablePixelFormat" error as in ggez. I'm not sure exactly why, but I believe that the HD4600 OpenGL driver doesn't correctly support antialiasing. But please continue reading....

Now I went back to ggez and tried to disable antialiasing by setting NumSamples to zero (glutin uses 0 to disable antialiasing). I then saw there was no zero option, and realized that ggez's 'NumSamples::One' means 'multisampling disabled'. However, it seems that using this value somehow enabled antialising.

Looking closely at the glutin documentation, it says you must set numsamples to '0' to disable multisampling - even when numsamples is '1', it enables multisampling (I assume). After looking at ggez's source code I concluded this was why antialiasing was being enabled when it shouldn't be - '1' was being passed in, which enables antialiasing.

## Testing on 0.4

I also went back to 0.4 to do some tests. To my surprise, any multisampling level allowed the program to 'run', although I don't know whether antialiasing is actually enabled or not. I'm guessing SDL2 just gets the 'best available' format, rather than erroring-out.

## Conclusion

So I'm not ENTIRELY sure that I'm correct but my 'findings' are:
- SDL2 will pick the best available pixel format if the one you want is not available (or at least, if the multisample part is not available)
- glutin will error-out if the pixel format is not available
- [glutin requires multisampling to be set to 0 to disable antialiasing](https://docs.rs/glutin/0.19.0/glutin/struct.ContextBuilder.html#method.with_multisampling). A setting of 1 still enables antialiasing.

In other words, "NumSamples::One" should actually pass '0' to glutin (or add a NumSamples::Disabled option), when `.with_multisampling()` is called here: https://github.com/ggez/ggez/blob/f0f51ae7c8850a08ace1eb3f665150553b92a583/src/graphics/context.rs#L113

EDIT: oh...just saw this comment:

https://github.com/ggez/ggez/issues/152#issuecomment-331204154

> I lean towards defaulting to 1x because it means that, well, nobody will ever have the examples fail to run with a "cannot create video context with the given settings" type error.

**Comment by icefoxen** [comment]
Wow, thank you for the exhaustive report and the research that went into it!  I've added a `NumSamples::Zero` which is now the default, since the default should be "no multisampling".  Like you, I thought that was what passing 1 did, but apparently not!  The changes have been pushed to `master`, do you mind testing them on your HD4600 and seeing whether it works?

**Comment by drojf** [comment]
Everything seems to work as expected (using the latest from the git repo):

- Default, without setting pixel format: Runs OK
- Set to Zero: Runs OK
- Set to One: No Pixel Format error

👍  I'll close the issue now
